cardano-cli/src/Cardano/CLI/Run/Hash.hs

```diff
@@ -0,0 +1,70 @@
+
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Cardano.CLI.Run.Hash
+  ( runHashCmds
+  ) where
+
+import           Cardano.Api
+import qualified Cardano.Api.Ledger as L
+
+import qualified Cardano.CLI.Commands.Hash as Cmd
+import           Cardano.CLI.Read
+import           Cardano.CLI.Types.Errors.HashCmdError
+import           Cardano.Crypto.Hash (hashToTextAsHex)
+
+import qualified Data.ByteString as BS
+import           Data.Function
+import qualified Data.Text.Encoding as Text
+import qualified Data.Text.IO as Text
+
+runHashCmds :: ()
+  => Cmd.HashCmds
+  -> ExceptT HashCmdError IO ()
+runHashCmds = \case
+  Cmd.HashAnchorDataCmd args -> runHashAnchorDataCmd args
+  Cmd.HashScriptCmd     args -> runHashScriptCmd     args
+
+runHashAnchorDataCmd :: ()
+  => Cmd.HashAnchorDataCmdArgs
+  -> ExceptT HashCmdError IO ()
+runHashAnchorDataCmd Cmd.HashAnchorDataCmdArgs { toHash, moutFile } =
+  case toHash of
+    Cmd.AnchorDataHashSourceBinaryFile fp -> do
+      let path = unFile fp
+      bytes <- handleIOExceptT (HashReadFileError path) $ BS.readFile path
+      let hash = L.hashAnchorData $ L.AnchorData bytes
+      printHash hash
+    Cmd.AnchorDataHashSourceTextFile fp -> do
+      let path = unFile fp
+      text <- handleIOExceptT (HashReadFileError path) $ Text.readFile path
+      let hash = L.hashAnchorData $ L.AnchorData $ Text.encodeUtf8 text
+      printHash hash
+    Cmd.AnchorDataHashSourceText text -> do
+      let hash = L.hashAnchorData $ L.AnchorData $ Text.encodeUtf8 text
+      printHash hash
+  where
+    printHash :: L.SafeHash L.StandardCrypto i -> ExceptT HashCmdError IO ()
+    printHash hash = do
```

Comment:
```suggestion
    writeHash :: L.SafeHash L.StandardCrypto i -> ExceptT HashCmdError IO ()
    writeHash hash = do
```
`print` would suggest printing to stdout, I think `write` would suit here better